include/gridtools/common/cuda_runtime.hpp

```diff
@@ -11,6 +11,10 @@
 
 #if defined(__HIP__)
 #include <hip/hip_runtime.h>
+#ifdef NDEBUG
```

Comment:
To be honest, I don't understand why this is not noop. What happens without defining `assert` on our own?

Reply:
It’s just a workaround for a HIP bug: HIP defines assert independently of `NDEBUG`, leading to compilation errors in our code (as it tries to access some stuff in release mode that is only defined when `NDEBUG` is present). But I actually forgot to file a HIP bug, so thanks for the reminder!